backend/execution_cache/src/mutation.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ArgoWorkflowTemplate string = "workflows.argoproj.io/template"
+	ExecutionKey         string = "executionKey"
+	AnnotationPath       string = "/metadata/annotations"
+)
+
+const (
+	Add OperationType = "add"
+)
+
+var (
+	podResource = metav1.GroupVersionResource{Version: "v1", Resource: "pods"}
+)
+
+// applyPodOutput will check whether the execution has already been run before from MLMD and apply the output into pod.metadata.output
+func applyPodOutput(req *v1beta1.AdmissionRequest) ([]patchOperation, error) {
+	// This handler should only get called on Pod objects as per the MutatingWebhookConfiguration in the YAML file.
+	// However, if (for whatever reason) this gets invoked on an object of a different kind, issue a log message but
+	// let the object request pass through otherwise.
+	if req.Resource != podResource {
+		log.Printf("expect resource to be %q", podResource)
+		return nil, nil
```

Comment:
What happens with the admission request when this line is hit?

Reply:
If the patchOperation is Nil, then there will be no patch operation happens. The pod gets executed without appending anything.